solr/core/src/java/org/apache/solr/servlet/CoordinatorHttpSolrCall.java

```diff
@@ -160,28 +156,13 @@ protected String getCoreOrColName() {
     return collectionName;
   }
 
-  public static SolrQueryRequest wrappedReq(
-      SolrQueryRequest delegate, String collectionName, HttpSolrCall httpSolrCall) {
-    Properties p = new Properties();
-    if (collectionName != null) {
-      p.put(CoreDescriptor.CORE_COLLECTION, collectionName);
-    }
-    p.put(CloudDescriptor.REPLICA_TYPE, Replica.Type.PULL.toString());
-    p.put(CoreDescriptor.CORE_SHARD, "_");
+  public static SolrQueryRequest wrappedReq(SolrQueryRequest delegate, HttpSolrCall httpSolrCall) {
```

Comment:
I believe this method can be removed altogether, can't it?

Reply:
Oh, I think I eventually got it! With coordinator nodes, there is a single synthetic core created per config set, hence service multiple collections. This method is here so make it able to service different collections by injecting an ad-hoc cloud descriptor. However, this seems like a very dodgy pattern, as calls going to `getCore().getCoreDescriptor().getCloudDescriptor()` would not work.

Reply:
Yup, that's what's happening and the rationale.
It would be hard to prevent this as most handler and related things like SearchComponents can be "SolrCoreAware" or otherwise manage to get access to the SolrCore via SolrQueryRequest.getCore.  That said, I think it'd be unusual for a handler or SearchComponent to do something erroneously with this access when in the coordinator mode.